src/parser/ModuleParser.mjs

```diff
@@ -91,13 +91,28 @@ export class ModuleParser extends StatementParser {
   // ImportSpecifier :
   //   ImportedBinding
   //   IdentifierName `as` ImportedBinding
+  //
+  // (*arbitrary-module-namespace-names) ImportSpecifier :
+  //   ImportedBinding
   //   ModuleExportName `as` ImportedBinding
   parseImportSpecifier() {
     const node = this.startNode();
-    if (this.feature('arbitrary-module-namespace-names') && this.test(Token.STRING)) {
-      node.ModuleExportName = this.parseModuleExportName();
-      this.expect('as');
-      node.ImportedBinding = this.parseBindingIdentifier();
+    if (this.feature('arbitrary-module-namespace-names')) {
```

Comment:
this is merged into the spec now, so you can remove the flag instead of duplicating all this code